gui/mass-remove.lua

```diff
@@ -0,0 +1,336 @@
+--building/construction mass removal/suspension tool
+
+--[====[
+
+gui/mass-remove
+===============
+Allows removal of buildings/constructions and suspend/unsuspend using
+a box selection.
+]====]
+
+local gui = require "gui"
+local guidm = require "gui.dwarfmode"
+local persistTable = require 'persist-table'
+
+MassRemoveUI = defclass(MassRemoveUI, guidm.MenuOverlay)
+
+--[====[
+The actions that can be taken. Key bindings in parentheses.
+
+--suspend (s): suspends the construction of a planned building/construction
+--unsuspend (p): resumes the construction of a planned building/construction
+--remove_n (n): designates a construction (wall, floor, etc) for removal. Similar to the native Designate->Remove Construction menu in DF
+--unremove_n (c): cancels removal of a construction (wall, floor, etc)
+--remove_x (x): designates a building (door, workshop, etc) for removal. Similar to the native Set Building Tasks/Prefs->Remove Building menu in DF
+--unremove_x (b): cancels removal of a building (door, workshop, etc)
+--remove_a (a): designates both constructions and buildings for removal, and deletes planned buildings/constructions
```

Comment:
Could you move this into the first block comment above? That would include it in the user-facing docs; it's not visible there currently, and I think it's useful enough to be there.

It would also need to be formatted as RST in that case. I think the only changes needed would be to
- replace `--` with `- ` (that includes a trailing space)
- replace the internal names with human-readable names (e.g. `remove_n` -> `Remove`), ideally ones that match the UI

...but I can help with any syntax issues that come up. https://docs.dfhack.org/en/stable/docs/Documentation.html explains how to build the docs, but it doesn't work well if you don't also have the DFHack repo cloned.

Reply:
Absolutely! I got this re-formatted and moved up, and we can check to see if it builds the docs right. I got the DFHack repo cloned on my machine, but not CMake. Would just running Sphinx work in this case to check if it builds them right?

Reply:
Whoops, sorry for the delay. Yes, `docs/build.sh` from the DFHack repo should work (on non-Windows), or you can run Sphinx manually - https://docs.dfhack.org/en/latest/docs/Documentation.html#building-the-documentation has more instructions.

Reply:
It looks like this, for reference (and it builds warning-free).

The keys did get out of sync, though (`p` changed to `Shift-s`, for instance) - It's probably fine to just remove these because they're on-screen when the script is running.

![image](https://user-images.githubusercontent.com/3719547/112092862-cbfdbf80-8b6e-11eb-8c5a-a5335bac922d.png)